src/test/java/ttt/PromptSpy.java

```diff
@@ -48,9 +50,14 @@ public int getNumberOfTimesDrawMessageHasBeenPrinted() {
     public String getLastBoardThatWasPrinted() {
         StringBuilder gridFormation = new StringBuilder();
 
-        for (int i = 0; i < BOARD_DIMENSION * BOARD_DIMENSION; i++) {
-            gridFormation.append(lastBoardPrinted.getSymbolAt(i));
+        Cell[][] rows = lastBoardPrinted.getRows();
+
+        for(Cell[] cells: rows) {
+            for(int i = 0; i < cells.length; i++) {
+                gridFormation.append(lastBoardPrinted.getSymbolAt(cells[i].getOffset()));
+            }
         }
+
```

Comment:
Have you considered using foreach here?

```
for (Cell cell : cells) {
    gridFormation.append(lastBoardPrinted.getSymbolAt(cell.getOffset()));
}
```

IntelliJ does this with one `option+enter` command.